jmxtrans-output/jmxtrans-output-gelf/src/test/java/com/googlecode/jmxtrans/model/output/gelf/GelfWriterTests.java

```diff
@@ -0,0 +1,337 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.googlecode.jmxtrans.model.output.gelf;
+
+import com.google.common.collect.ImmutableList;
+import com.googlecode.jmxtrans.model.Result;
+import com.googlecode.jmxtrans.model.ResultFixtures;
+import com.googlecode.jmxtrans.model.ValidationException;
+import org.graylog2.gelfclient.GelfMessage;
+import org.graylog2.gelfclient.transport.GelfTcpTransport;
+import org.graylog2.gelfclient.transport.GelfTransport;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentMatcher;
+import org.mockito.Mockito;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.googlecode.jmxtrans.model.QueryFixtures.dummyQuery;
+import static com.googlecode.jmxtrans.model.ServerFixtures.dummyServer;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+import static org.mockito.Matchers.argThat;
+import static org.mockito.Matchers.isA;
+import static org.mockito.Mockito.doNothing;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+public class GelfWriterTests {
+
+	private Result result;
+
+	@Before
+	public void setUp() throws Exception {
+		this.result = ResultFixtures.numericResult();
+	}
+
+	@Test
+	public void testSendGelfMessage() throws Exception {
+		final ImmutableList<String> typenames = ImmutableList.of();
+		final Map<String, Object> settings = new HashMap<>();
+
+		final Map<String, Object> additionalFields = new HashMap<>();
+
+		additionalFields.put("test", "test");
+
+		final GelfTransport gelfTransport = Mockito.mock(GelfTcpTransport.class);
+
+		doNothing().when(gelfTransport).send(
+			argThat(
+				new ArgumentMatcher<GelfMessage>() {
+					@Override
+					public boolean matches(final Object o) {
+						final GelfMessage message = (GelfMessage) o;
+						assertThat(message.getAdditionalFields())
+							.containsKey("test")
+							.as("Additional fields were not checked");
+
+						assertThat(message.getFullMessage())
+							.isNotEmpty()
+							.as("Invalid full message");
+						assertThat(message.getMessage())
+							.isNotEmpty()
+							.as("Invalid message");
+						return true;
+					}
+				}
+			)
+		);
+
+		final GelfWriter gelfWriter;
+
+		gelfWriter = new GelfWriter(
+			typenames,
+			true,
+			true,
+			settings,
+			additionalFields,
+			gelfTransport
+		);
+		gelfWriter.start();
+		gelfWriter.doWrite(dummyServer(),
+			dummyQuery(),
+			ImmutableList.of(result));
+
+		verify(
+			gelfTransport,
+			times(1)
+		).send(isA(GelfMessage.class));
+	}
+
+	@Test
+	public void testTcpDefaultsWriterFactory() throws ValidationException {
+
+		final ImmutableList<String> typenames = ImmutableList.of();
+		final Map<String, Object> settings = new HashMap<>();
+
+		final Map<String, Object> additionalFields = new HashMap<>();
+
+		additionalFields.put("test", "test");
```

Comment:
Adding additional fields to test that they are working correctly is good! Testing that they are actually sent would be much better. You should probably create a mock transport, [capture](https://static.javadoc.io/org.mockito/mockito-core/2.8.47/org/mockito/Mockito.html#15) the message and do a few assertions on it. The same kind of additional assertions could be added to most of those tests.

Side note: [ImmutableMap.of()](https://google.github.io/guava/releases/20.0/api/docs/com/google/common/collect/ImmutableMap.html#of-K-V-) would be a much shorter notation to declare those `additionalFields`.